tests/infra/clients.py

```diff
@@ -261,13 +265,13 @@ def __init__(
         self.format = format
 
         while connection_timeout >= 0:
+            connection_timeout -= 0.1
```

Comment:
This loop did not work properly. When `connection_timeout` was about to be decremented for the last time (i.e. `connection_timeout=0`) and `self.connect()` was failing, the exception would be caught, `connection_timeout` was still positive and we wouldn't `raise`. We would then decrement `connection_timeout` (now equals to `-0.1`) and the `while` loop would stop. We would then try to send a request, which would timeout "forever" (`ctest` actually times out after 240 seconds).

Reply:
Ah, good catch!